File: benchmarks/nqueens/implementations/matlab/runner.m
function runner(size)
	tic;
	[solutions,us] = nqueen_cpu(size);
	elapsedTime = toc;

	fprintf(1, '{ "status": 0, "options": "runner(%s)", "time": %f,  "output": "[%d, %d]" }\n', num2str(size), elapsedTime, solutions, us);
end

File: benchmarks/nqueens/implementations/js/runner.js
if (typeof performance === 'undefined') {
  var performance = Date
}

function run (size) {
  var us = {}
  var t1, t2

  var solutions

  t1 = performance.now()
  solutions = nqueenJS(size, us)
  t2 = performance.now()

  console.log('Size: ' + size + ' Time: ' + (t2 - t1) / 1000 + ' s Solutions: ' +
    solutions + ' Unique Solutions: ' + us['solutions'])
  console.log(JSON.stringify({
    status: 1,
    options: 'run(' + size + ')',
    time: (t2 - t1) / 1000,
    output: '[' + solutions + ', ' + us['solutions'] + ']'
  }))
  server.done()
}
